lumenlearning/lti-middleware
# Dl3 18 pair lumen course to lms course endpoint, fetch links from harmony, submit links to LMS

## Description
- Pairs the LMS course context with the Lumen book (root_outcome_guid)
- Fetches the links from Harmony
- Submits the links to the LMS when the instructor clicks Add Course

### Motivation and Context
 - This pairs the LMS course context with the root_outcome_guid so that checks can be made upon subsequent launches of the Deep Linking menu to determine if the course has already been paired or not to determine what to display to the user.
 - This fetches the links from Harmony and submits them to the LMS to complete the initial Deep Linking flow to generate useable LTI Resource Links within the instructor's LMS course.

### Fixes
[DL3-18](https://lumenlearning.atlassian.net/browse/DL3-18)

## How Has This Been Tested?
1. Ensure that a course is setup in the LMS and that the LTI tool has been configured in the LMS for Deep Linking.
2. Go to the LMS course and trigger the opening of the deep linking menu.
3. Select a course from the menu.
4. Click the "Add Course" button.
5. Wait for the LMS to reload.
6. Note that the links have been added to the course and that when clicked on, they go through the LTI 1.3 Resource Link flow.
7. Note that the assignments appear in the gradebook.

## Screenshots
N/A
---
## Database changes
N/A

## ⚠️ Deployment instructions ⚠️
N/A

## New ENV variables
N/A

---

## Checklist
